packages/flutter_tools/gradle/src/main/groovy/flutter.groovy

```diff
@@ -824,25 +818,17 @@ class FlutterPlugin implements Plugin<Project> {
      * TODO: Remove this or compareVersionStrings. This does not handle strings like "8.6-rc-2".
      */
     static String mostRecentSemanticVersion(String version1, String version2) {
-        List version1Tokenized = version1.tokenize(".")
-        List version2Tokenized = version2.tokenize(".")
-        int version1numTokens = version1Tokenized.size()
-        int version2numTokens = version2Tokenized.size()
-        int minNumTokens = Math.min(version1numTokens, version2numTokens)
-        for (int i = 0; i < minNumTokens; i++) {
-            int num1 = version1Tokenized[i].toInteger()
-            int num2 = version2Tokenized[i].toInteger()
-            if (num1 > num2) {
-                return version1
-            }
-            if (num2 > num1) {
-                return version2
+        List<Integer> version1Tokenized = version1.tokenize(".")*.toInteger()
```

Comment:
Semantically I don't think this is the same. Specifically I think a number like 8.6-rc-2 would crash on the first 2 lines.

Reply:
yep it should be `List<String>` but still this does not resolve cases like `8.6-rc-2` , like the TODO above says. 
hmmm , should i work for an overall fix for this or just let it be list<String> ?

Reply:
Most of this pr is good, lets remove this change and land the rest. Then you can either tackle this or not based on what you prefer.

Reply:
currenlty i'm seeing [a list of all gradle versions ](https://services.gradle.org/distributions/) (3160 as of today) ,i see that there is only the special case of `rc` , but i have no idea how to compare them for exemple is `8.7-rc-1` with `8.8` , i guess 8.8 is newer

Reply:
well i found some script , tweaked it a bit and it seam to wrok :
(could u kindly confirm that this the intended behavior ?)
```groovy
def compareGradleVersions(String version1, String version2) {

    def v1Parts = version1.tokenize('.-')
    def v2Parts = version2.tokenize('.-')

    
    for (int i = 0; i < Math.max(v1Parts.size(), v2Parts.size()); i++) {
        
        def v1Part = i < v1Parts.size() ? v1Parts[i] : '0'
        def v2Part = i < v2Parts.size() ? v2Parts[i] : '0'

        
        def v1Num = v1Part.isNumber() ? v1Part.toInteger() : 0
        def v2Num = v2Part.isNumber() ? v2Part.toInteger() : 0

        if (v1Num != v2Num) {
            return v1Num > v2Num ? version1 : version2
        }

        // like -rc-1 cass
        if (v1Part.isNumber() && !v2Part.isNumber()) {
            return version1
        } else if (!v1Part.isNumber() && v2Part.isNumber()) {
            return version2
        } else if (v1Part != v2Part) {
            return comparePreReleaseIdentifiers(v1Part, v2Part) ? version1 : version2
        }
    }
    return version1 
    // If both are same return one of thm 
}

def comparePreReleaseIdentifiers(String v1Part, String v2Part) {
    // i tested "rc-1" vs "rc-2" worked (rc-2)
    // i tested "8.7-rc-1" vs "8.7" worked (8.7)
    // i tested "8.7-rc-1" vs "8.7.2" worked (8.7.2)
    // i tested "8.7.1" vs ""8.7.2"" worked (8.7.2)
    def v1PreRelease = v1Part.replaceAll("\\D", "")
    def v2PreRelease = v2Part.replaceAll("\\D", "")
    
    return v1PreRelease < v2PreRelease
}


def newerVersion = compareGradleVersions("8.7.1", "8.7.2")
println "The newer version is: $newerVersion"

```